COMET/misc_plugins/PlotScripts/analysis_scripts/IV_PQC.py

```diff
@@ -1,82 +1,670 @@
+            Path_Isurf = hv.Arrow(-1, mxx, 'max', '^')
+            Path_Isurf_average = hv.Arrow(0, I_surf_maxima_average, 'average', '^')
+            # Plot all Measurements
+            self.donts_gatediode = ["timestamp", "voltage", "Voltage", 'Current', "Stepsize", "Wait", "Stepsize", "Frequency", "x", "N"]  # don't plot these.
+            self.PlotDict["BasePlots_gate"] = plot_not_kink
+            self.PlotDict["BasePlots_gate"] += dif_intep_plot
+            self.PlotDict["BasePlots_gate"] += dif2_intep_plot
+            self.PlotDict["BasePlots_gate"] += curr_interp_plot
+            try:
+                self.PlotDict["BasePlots_gate"] += curr_savgol_plot
+                self.PlotDict["BasePlots_gate"] += curr_savgol_plot * plot_not_kink
+            except Exception:
+                self.log.warning("No savgol plot possible... Error: {}")
+            self.PlotDict["BasePlots_gate"] += text * curr_interp_plot * plot_not_kink * Path_min * Path_mxx * Path_average * Path_Isurf * Path_Isurf_average * Path_savgolmax
+            self.PlotDict["BasePlots_gate"] += curr_interp_plot * dif_intep_plot * dif2_intep_plot * plot_not_kink
+
+            # Add table that shows resulting parameters of the analysis
+            df4 = pd.DataFrame({"Name": gate_files, "Surface current_max (A)": Surface_current,
+                                "Surface current_average (A)": Surface_current_average,
+                                'Surface recombination velocity_max (cm/s)': Surface_recombination_velocity,
+                                'Surface recombination velocity_average (cm/s)': Surface_recombination_velocity_average})
+            table3 = hv.Table((df4), label='Gate analysis')
+            table3.opts(width=1300, height=800)
+            self.PlotDict["BasePlots_gate"] += table3
+
+        # Do this if the analysis is of more than one file
+        elif len(gate_files) > 1:
+            self.donts = ["timestamp", "voltage", "Voltage", "Stepsize", "Wait", "Stepsize", "Frequency", "firstderivative_gate", "x", "N"]  # do not plot this data
+            self.basePlots3 = plot_all_measurements(self.data, self.config, self.xaxis, self.name, do_not_plot=self.donts, keys=gate_files)
+            self.PlotDict["BasePlots_gate"] = self.basePlots3
+
+            # Add table that shows resulting parameters of the analysis
+            df4 = pd.DataFrame({"Name": gate_files, "Surface current_max (A)": Surface_current,
+                                "Surface current_average (A)": Surface_current_average,
+                                'Surface recombination velocity_max (cm/s)': Surface_recombination_velocity,
+                                'Surface recombination velocity_average (cm/s)': Surface_recombination_velocity_average})
+            table3 = hv.Table((df4), label='Gate analysis')
+            table3.opts(width=1300, height=800)
+            self.PlotDict["BasePlots_gate"] += table3
+
+        return self.PlotDict["BasePlots_gate"]
+
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/dominic.bloech/UniDAQ/pullRequest?prid=5481807)